### PR TITLE
minidlna: 1.3.0 -> 1.3.1

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change
Prevents DNS rebinding attacks through malicious remote web servers.

https://www.openwall.com/lists/oss-security/2022/03/03/1

Fixes: CVE-2022-26505

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/minidlna/default.nix [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
+  inherit pname version;
 
-  src = fetchurl {
-    url = "mirror://sourceforge/project/minidlna/minidlna/${version}/minidlna-${version}.tar.gz";
-    sha256 = "0qrw5ny82p5ybccw4pp9jma8nwl28z927v0j2561m0289imv1na7";
+  # tarball for 1.3.1 is missing
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/${pname}/git";
+    rev = "v${builtins.replaceStrings [ "."] [ "_" ] version}";
+    hash = "sha256-nbvz/QHSZBTZEqX/utOoOF5vorhrxGqIBA9qfpIZzyU=";
```
Comment (alyssais) [inline pkgs/tools/networking/minidlna/default.nix:14]: Sourceforge will give us git snapshot tarballs: https://sourceforge.net/code-snapshots/git/m/mi/minidlna/git.git/minidlna-git-66939d5ac4986020764c0518455f55b68422369f.zip

Maybe we need a `fetchFromSourceForgeGit`…

Comment (alyssais) [reply pkgs/tools/networking/minidlna/default.nix:14]: Actually, apparently Sourceforge snapshots are short-lived, so probably best not to use snapshot URLs without a dedicated fetcher that ensures they're regenerated before downloading.

File: pkgs/tools/networking/minidlna/default.nix [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
+stdenv.mkDerivation {
+  inherit pname version;
 
-  src = fetchurl {
-    url = "mirror://sourceforge/project/minidlna/minidlna/${version}/minidlna-${version}.tar.gz";
-    sha256 = "0qrw5ny82p5ybccw4pp9jma8nwl28z927v0j2561m0289imv1na7";
+  # tarball for 1.3.1 is missing
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/${pname}/git";
+    rev = "v${builtins.replaceStrings [ "."] [ "_" ] version}";
```
Comment (alyssais) [inline pkgs/tools/networking/minidlna/default.nix:13]: [nit] Missing space here.

```suggestion
    rev = "v${builtins.replaceStrings [ "." ] [ "_" ] version}";
```

Comment (mweinelt) [reply pkgs/tools/networking/minidlna/default.nix]: Thanks, fixed!

**Comment by risicle** [comment]
@ofborg test minidlna

This fails for me, though it passes on master.

**Comment by mweinelt** [comment]
The webserver replies just fine, it just answers with a 404, so the tests expects something which isn't there.

**Comment by mweinelt** [comment]
Alternative course of action could be to just apply https://sourceforge.net/p/minidlna/git/ci/c21208508dbc131712281ec5340687e5ae89e940/.

**Comment by risicle** [comment]
I have a feeling that patch may have done it itself. It works by rejecting any request that has a `Host` header with any character not in the set `[.:0-9]`, i.e. an IPv4 address. And our curl requests will presumably use a `Host` header of either `localhost` or `server`.

**Comment by risicle** [comment]
`curl --fail http://$(getent hosts server | head -n1 | cut -f 1 -d ' '):8200/` succeeds for `server` but for `localhost` my guess is it's giving us an ipv6 address and I don't know if we can stop that with `getent`. `host` could be the answer...

**Comment by risicle** [comment]
Ok I have a fix, will push...

**Pull request merged**